crates/environ/src/types.rs

```diff
@@ -1020,47 +1043,47 @@ impl fmt::Display for WasmSubType {
 impl WasmSubType {
     #[inline]
     pub fn is_func(&self) -> bool {
-        self.composite_type.is_func()
+        self.composite_type.inner.is_func()
```

Comment:
For this accessor and those below I think we'll want to implicitly redefine `is_func` as `is_unshared_func` (although without the renaming for now). Eventually if needed we can add `is_shared_func` but otherwise these should all get extra boolean conditions and such to check if `self.shared` and act appropriately (e.g. return `false`, `None`, or `panic!`)